code/controllers/subsystem/processing/circuit.dm

```diff
@@ -0,0 +1,77 @@
+PROCESSING_SUBSYSTEM_DEF(circuit)
+	name = "Circuit"
+	stat_tag = "CIR"
+	var/list/all_exonet_connections = list()				//Address = connection datum.
+	var/list/all_integrated_circuits = list()
+	var/list/obj/machinery/exonet_node/all_exonet_nodes = list()
+	var/cipherkey
+	var/list/circuit_fabricator_recipe_list = list()
+	init_order = INIT_ORDER_CIRCUIT
+	flags = NONE
+
+/datum/controller/subsystem/processing/circuit/Initialize(start_timeofday)
+	SScircuit.cipherkey = random_string(2000+rand(0,10), GLOB.alphabet)
+	initialize_integrated_circuits_list()
+	initialize_circuit_fabricator_recipes()
+	return ..()
+
+/datum/controller/subsystem/processing/circuit/proc/initialize_integrated_circuits_list()
+	all_integrated_circuits = list()
+	for(var/thing in typesof(/obj/item/integrated_circuit))
+		all_integrated_circuits += new thing()
```

Comment:
I don't want to have to figure out all uses of this so correct me if I'm wrong but it looks like you don't need to have instanced circuits populating this list.

Store type paths in place of the object and you can access the vars like the following:
```
var/obj/item/integrated_circuit/thing = /obj/item/integrated_circuit
var/var_value = initial(thing.var)
```

Reply:
Nope. Maybe it's a little bit crude solution, but it's used to get type,name,displayed name,size and complexity. More than type only.

Reply:
Wait. I can get initial vars by type this way?

Reply:
Wait. I can get initial vars by type this way?

Reply:
yes, thats a thing.

Reply:
It will take time. hard to fix it.